### ISSUE TITLE
Handle oneOf/unions in schemas

### ISSUE THREAD

**Issue by SachaG** [VulcanJS/Vulcan#2539]
It would be nice to have the ability to handle unions in schemas. For example:

```js
const fooSchema = {
  bar: {
    type: SimpleSchema.oneOf(String, Number)
  }
}
```

would become

```graphql
type Foo {
  union bar: String | Number
}
```

And similarly with custom nested schemas:

```js
const fooSchema = {
  bar: {
    type: SimpleSchema.oneOf(createSchema(schema1), createSchema(schema2))
  }
}
```

Assuming we could get this to work (@eric-burel I looked at the code but am not familiar enough with it to know where/how to implement this…) the next step would be getting the forms to work with these union schemas. I guess we would need an extra property on the field that indicates which schema to pick somehow?

```js
const fooSchema = {
  bar: {
    type: SimpleSchema.oneOf(createSchema(schema1), createSchema(schema2)),
    typeSelect: document => document.category === 's1' ? createSchema(schema1) : createSchema(schema2)
  }
}
```

**Comment by eric-burel** [comment]
It shouldn't be too hard (everything happens in `schemaFields.js`, it's just new case) but I have questions about mutations, forms, and default fragments.

We can't have union inputs so right now I am not sure of the expected graphql for mutations including those fields. 

For fragment, I think the syntax is `...on UnionType1 { whatever } ...on UnionType2 { foobar }`? What happens if you mix types with subfields and primitives, like a string union an object (for example if you want to store either the object id or the actual object)?

For forms you'll need to have some pattern, for example 2 fields, one to select the type one to select the actual value (or tabs or whatever depending the size of the union).

**Comment by SachaG** [comment]
Yeah I guess I hadn't thought about mutations… See https://github.com/binaryseed/graphql-spec/blob/3532cc6edc1cde132ad35e4ff9b1e9ed105a03f0/rfcs/InputUnion.md (discussion here: https://github.com/graphql/graphql-spec/pull/618/files)

In that case should we leave this off until this gets resolved in the GraphQL spec? I guess I'll find another solution for my survey use case.

**Comment by eric-burel** [comment]
We don't necessarily have to wait, we may figure alternative.  For example using a `JSON` for the input type could be sufficient already. It's ok if we lose graphQL validation at this point imo, since the server already handles it.

 It could take too much time to implement for your own use case however.


Also adding a link for union of scalars, because the RFC only seem to talk about union of objects: https://github.com/graphql/graphql-spec/issues/215

I have no idea to handle non objects though because there is no default scalar type like JSON for object. We could add our own `any` type with a smart decoder that differentiate string, number, json and date for example.
